src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -1218,12 +1216,20 @@ void DoorLockServer::clearYearDayScheduleCommandHandler(chip::app::CommandHandle
     emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_SUCCESS);
 }
 
-bool DoorLockServer::HasFeature(chip::EndpointId endpointId, DoorLockFeature feature)
+bool DoorLockServer::HasFeatures(chip::EndpointId endpointId, uint32_t features)
```

Comment:
Could we use C++-safe types instead of EMBER_AF constants?

we should have `DoorLock::DoorLockFeature` type and a BitFlags of them.

Then below instead of using EMBER_AF_DOOR_LOCK_FEATURE_PIN_CREDENTIALS you would use `DoorLockFeature::kPINCredentials` which makes it typesafe.

Reply:
Using `chip::BitFlags` and adding `GetFeature` function will fix this also.